spring-integration-redis/src/test/java/org/springframework/integration/redis/store/RedisMessageGroupStoreTests.java

```diff
@@ -400,16 +403,21 @@ void testJsonSerialization() {
 		Message<?> mutableMessage = new MutableMessage<>(UUID.randomUUID());
 		Message<?> adviceMessage = new AdviceMessage<>("foo", genericMessage);
 		ErrorMessage errorMessage = new ErrorMessage(new RuntimeException("test exception"), mutableMessage);
+		Message<?> delayMessage = new DefaultMessageBuilderFactory()
+				.withPayload(new DelayHandler.DelayedMessageWrapper(genericMessage, System.currentTimeMillis()))
+				.copyHeaders(genericMessage.getHeaders())
+				.build();
 
-		store.addMessagesToGroup(this.groupId, genericMessage, mutableMessage, adviceMessage, errorMessage);
+		store.addMessagesToGroup(this.groupId, genericMessage, mutableMessage, adviceMessage, errorMessage, delayMessage);
 
 		MessageGroup messageGroup = store.getMessageGroup(this.groupId);
-		assertThat(messageGroup.size()).isEqualTo(4);
+		assertThat(messageGroup.size()).isEqualTo(5);
 		List<Message<?>> messages = new ArrayList<>(messageGroup.getMessages());
 		assertThat(messages.get(0)).isEqualTo(genericMessage);
 		assertThat(messages.get(0).getHeaders()).containsKeys(MessageHistory.HEADER_NAME);
 		assertThat(messages.get(1)).isEqualTo(mutableMessage);
 		assertThat(messages.get(2)).isEqualTo(adviceMessage);
+		assertThat(messages.get(4)).isEqualTo(delayMessage);
```

Comment:
This is not very logic order.
Why cannot we verify `delayMessage` after `errorMessage`?